lib/reporters/base.js

```diff
@@ -211,6 +211,56 @@ var generateDiff = (exports.generateDiff = function (actual, expected) {
   }
 });
 
+/**
+ * Traverses err.cause and returns all stack traces
+ *
+ * @private
+ * @param {Error} err
+ * @param {Set<Error>} [seen]
```

Comment:
[Docs] Since there's an `if (seen`:

```suggestion
 * @param {Set<Error> | undefined} [seen]
```

Reply:
When using eg. [`exactOptionalPropertyTypes`](https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes) with TypeScript there is a distinct difference between the two.

With your change both `getFullErrorStack(err, undefined)` and `getFullErrorStack(err, undefined)` is allowed, with the original code only `getFullErrorStack(err)` is allowed.

Within the function it will always be seen as possibly being `undefined` (at least if one is using `strictNullChecks` when validating, but nothing is validated in Mocha yet ;P)

Reply:
Never mind then 😄